arwen/contexts/runtime.go

```diff
@@ -67,6 +74,16 @@ func NewRuntimeContext(
 		useDifferentGasCostForReadingCachedStorageEpoch: useDifferentGasCostForReadingCachedStorageEpoch,
 	}
 
+	var err error
+	context.warmInstanceCache, err = lrucache.NewCacheWithEviction(warmCacheSize, instanceEvicted)
```

Comment:
I would have used only one cache containing both the instance pointer and the local data in a specialized, unexported struct. 
The reasons for this are the following:
1. in no way we could have the situation in which a local data value is evicted async with the instance (requiring the recreation of the warm instance). 
2. less type assertions as the only one used is between the cached object from the cache and the unexported struct containing both the local data and the instance pointer.

Reply:
good idea :). will do.

Reply:
done